pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/documentation/ApexDocRule.java

```diff
@@ -106,6 +128,11 @@ public Object visit(ASTProperty node, Object data) {
         return data;
     }
 
+    protected void init() {
```

Comment:
You can override `start(RuleContext)` instead of using your custom init method. It will be called once per file. Even better, you could remove the fields and just use getProperty in the handleClassOrInterface method.

Reply:
Understood, I was misguided by following other rules approach such as the `VariableNamingConventionsRule` one